src/project/server/routes/api.js

```diff
@@ -0,0 +1,114 @@
+// Define your endpoints here (this is your "controller file")
+const express = require('express');
+const ItemManager = require('../services/item-manager.js')
+
+const todoRouter = express.Router();
+const itemManager = new ItemManager();
+
+todoRouter.get('/', async(req, res) => {
+    const {sort, filter} = req.query;
+    
+    let sortedList = [];
+
+    const sortObj = {
+        atoz:{field: 'itemName', order:'ASC'},
+        ztoa:{field: 'itemName', order:'DESC'},
+        dtou:{field: 'status', order:'DESC'},
+        utod:{field: 'status', order:'ASC'},
+    }
+
+    if(sort){
+        sortedList = await itemManager.orderData(sortObj[sort].field, sortObj[sort].order)
+    }
+    else if(filter){
+
+        if(filter === 'checked'){
+            sortedList = await itemManager.filterData(true)
+        }else{
+            sortedList = await itemManager.filterData(false)
+        }
+    }
+    else{
+        sortedList = await itemManager.getTodoList()
+    }
+    res.status(200).json(sortedList);
+})
+
+todoRouter.post("/", async(req, res) => {
+    const {todo} = req.body
+    let error = new Error()
+    
+    if(!todo){
+        error.statusCode = 400
+        error.message = "Invalid todo, todo is null"
+        throw error
+    }
+    
+    try{
+        await itemManager.addTodo(todo)
+        res.status(201).json(todo);
```

Comment:
it's a best practice to call return on res.json() so that it's easier to notice that this is the last thing that will run